signal_test.go

```diff
@@ -0,0 +1,95 @@
+package signal_test
+
+import (
+	"testing"
+
+	format "fmt"
+
+	"github.com/NeowayLabs/signal"
+)
+
+type testcase struct {
+	sig signal.Discrete
+	μ   float64 // mean
+	σ   float64 // std deviation
+	σ2  float64 // variance
+}
+
+var fmt = format.Sprintf
+var testcases = []testcase{
+	{sig: signal.Discrete{1, 1, 1, 1}, μ: 1, σ: 0},
+	{sig: signal.Discrete{1, 1, 1, 0}, μ: 0.75, σ: 0.5, σ2: 0.25},
+	{sig: signal.Discrete{1, 1, 0, 0}, μ: 0.5, σ: 0.577350269, σ2: 0.333333333},
+	{sig: signal.Discrete{1, 0, 0, 0}, μ: 0.25, σ: 0.5, σ2: 0.25},
+	{sig: signal.Discrete{0, 0, 0, 0}, μ: 0, σ: 0, σ2: 0},
+	{}, // division by zero
+	{
+		sig: signal.Discrete{
+			0, 1, 2, 3, 4, 5, 6, 7, 8, 9,
+			10, 20, 30, 40, 50, 60, 70, 80, 90, 100,
+			200, 300, 400, 500, 600, 700, 800, 900, 1000, 1100,
+			200, 300, 400, 500, 600, 700, 800, 900, 1000, 1100,
+			10, 20, 30, 40, 50, 60, 70, 80, 90, 100,
+			0, 1, 2, 3, 4, 5, 6, 7, 8, 9,
+		},
+		μ:  236.5,
+		σ:  340.030781258,
+		σ2: 115620.932203389,
+	},
+}
+
+func assert(t *testing.T, b bool, msg string) {
```

Comment:
hmmm t.Helper...fancy =P

Sometimes I feel like having a very small assert lib. For example I have some version of asserting that supports formatted errors somewhere, it was pretty cool. What you think ?

Reply:
A little assert lib could be useful :)

In this case, every assertX reused the same formatted message, then the benefit of formatting would be little. But yes, all this could be in our assert-stuff lib.

Reply:
Will do something on that direction =)